Issues/4035 update readerpost model

aerych opened the pull request:
Refs: #4035 
This PR: 
- Adds some new fields to the `ReaderPost` model
- removes some old obsolete fields and methods 
- updates the remote service to populate the new fields
- adds some tests for newly added remote service methods

Since we're still in a dev cycle I didn't bump the model revision for these changes. 

To test: 
- Since the new fields are not yet used confirm the app launches and the reader syncs and displays its contents without error.
- Run the test suite and confirm there are no errors

@SergioEstevao would you be up for a quick review? 

Needs Review: @SergioEstevao

SergioEstevao commented on WordPress/Classes/Networking/ReaderPostServiceRemote.m:
```diff
@@ -266,45 +287,121 @@ - (RemoteReaderPost *)formatPostDictionary:(NSDictionary *)dict
+                postCount = count;
+            }
+        }
+    }
```
If I'm reading this code right we only take in account two of the tags in the remoteTags lists? The last two? Is this expected/wanted?

aerych replied:
Sort of.  The logic should be:
- Loop over all the tags.
- If the current tag's post count is greater than the previous post count, make it the new primary tag, and make a previous primary tag the secondary tag. 

We're only concerned with the two tags with the highest post count.
I'll add a comment for clarity.

SergioEstevao commented on WordPress/Classes/Networking/ReaderPostServiceRemote.m:
```diff
@@ -266,45 +287,121 @@ - (RemoteReaderPost *)formatPostDictionary:(NSDictionary *)dict
+
+    NSInteger minutesToRead = count / avgWordsPerMinuteRead;
+    if (minutesToRead < minimumMinutesToRead) {
+        return @(0);
```
Shouldn't this be 2 instead of 0? Should the avWordPerMinuteToRead and minimumMinutesToRead be constants?

aerych replied:
It should be zero, but I think I could find a better name for `minimumMinuteToRead`. I think the name makes it confusing why we'd want a zero returned.
They should totally be constants.  I'll make that change :)

SergioEstevao commented:
@aerych  There are two unit tests that are failing but they don't look related to these changes. 
I tested the app and all is working ok so have a look on my other comments when you address them I will have another go.

aerych commented:
Thanks @SergioEstevao!  I've added constants and clarified a few things.  I also fixed the tests. I think one of them broke when I split the reader service tests into two files.   
Ready for another round.

SergioEstevao commented:
@aerych :shipit:

aerych commented:
Thanks @SergioEstevao!
